src/client/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryManager.cs

```diff
@@ -109,14 +108,26 @@ internal class InstanceDiscoveryManager : IInstanceDiscoveryManager
             RequestContext requestContext, 
             bool forceValidation = false)
         {
-
             Uri authorityUri = authorityInfo.CanonicalAuthority;
             string environment = authorityInfo.Host;
 
             if (authorityInfo.IsInstanceDiscoverySupported)
             {
                 var entry = _userMetadataProvider?.GetMetadataOrThrow(environment, requestContext.Logger);
 
+                //Check if instance discovery endpoint is disabled
+                if (entry == null && requestContext.ServiceBundle.Config.InstanceDiscoveryEndpoint == Constants.InstanceDiscoveryEndpointDisabled)
```

Comment:
You also need to update `GetMetadataEntryTryAvoidNetworkAsync` and have a test for this.

Reply:
Why? `GetMetadataEntryTryAvoidNetworkAsync` doesn't go over the network, it ends up calling this method.

Reply:
I think it does https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryManager.cs

Maybe the name or the design is wrong. i.e. "TryAvoid" is not the same as "Avoid". It avoids going to the network if all the hostnames in the cache == authority.host

Reply:
Yea, but `GetMetadataEntryTryAvoidNetworkAsync ` calls `GetMetadataEntryAsync` todo the network call, which already has the check.